// web3.storage/dist/src/platform.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _fetch = require('@web-std/fetch');
var blob = require('@web-std/blob');
var file = require('@web-std/file');
var fs = require('ipfs-car/blockstore/fs');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _fetch__default = /*#__PURE__*/_interopDefaultLegacy(_fetch);



Object.defineProperty(exports, 'Headers', {
  enumerable: true,
  get: function () { return _fetch.Headers; }
});
Object.defineProperty(exports, 'Request', {
  enumerable: true,
  get: function () { return _fetch.Request; }
});
Object.defineProperty(exports, 'Response', {
  enumerable: true,
  get: function () { return _fetch.Response; }
});
Object.defineProperty(exports, 'fetch', {
  enumerable: true,
  get: function () { return _fetch__default["default"]; }
});
Object.defineProperty(exports, 'Blob', {
  enumerable: true,
  get: function () { return blob.Blob; }
});
Object.defineProperty(exports, 'File', {
  enumerable: true,
  get: function () { return file.File; }
});
Object.defineProperty(exports, 'Blockstore', {
  enumerable: true,
  get: function () { return fs.FsBlockStore; }
});
